src/cclib/parser/gamessparser.py

```diff
@@ -68,13 +71,107 @@ def before_parsing(self):
         self.firststdorient = True  # Used to decide whether to wipe the atomcoords clean
         self.cihamtyp = "none"  # Type of CI Hamiltonian: saps or dets.
         self.scftype = "none"  # Type of SCF calculation: BLYP, RHF, ROHF, etc.
+        self.metadata['methods'] = [] #methods could have mutliple values
 
     def extract(self, inputfile, line):
         """Extract information from the file object inputfile."""
         
+        # extract the version number first
+        if line.find("GAMESS VERSION") >= 0:
+            self.metadata["package_version"] = line.split()[4]+line.split()[5]+line.split()[6]
+
         if line[1:12] == "INPUT CARD>":
             return
 
+        # extract the methods 
+        if line[1:7] == "SCFTYP":
+            method = line.split()[0][7:]
+            if len(self.metadata["methods"]) == 0:
+                self.metadata["methods"].append(method)
+
+        # extract the basis set name
+        if line[5:11] == "GBASIS":
+            basnm1 = line.split()[0][7:]
```

Comment:
I have built a dictionary for Dunning basis sets. It's difficult to do it for other basis sets because they involves several line contents.